### PR TITLE
Provider/Azure: Properly propagate scale-up failures in scale sets.

### PR DESCRIPTION
**Pull request by nilo19**
Properly propagate scale-up failures in scale sets and update the corresponding vendors.

Ref: https://github.com/kubernetes/autoscaler/pull/2944/files

/area provider/azure
/kind bug

### CODE REVIEW

**Comment by feiskyer** [comment]
@losipiuk could you help to approve the vendor updates?

**Comment by nilo19** [comment]
> This is still not propagating the errors in `SetScalesetSize` back to `IncreaseSize`

I've amended the commit. Now the` SetScaleSetSize` propagates the error.

File: cluster-autoscaler/cloudprovider/azure/azure_util_test.go [hunk 1]
```diff
@@ -21,8 +21,9 @@ import (
 	"net/http"
 	"testing"
 
-	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-12-01/compute"
```
Comment (marwanad) [inline cluster-autoscaler/cloudprovider/azure/azure_util_test.go]: Can we do the azure sdk updates in a separate PR - just to keep this diff small and manageable?

**Comment by losipiuk** [comment]
Please do not update vendor directly. Follow guidelines in https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-update-ca-dependencies-particularly-k8siokubernetes

**Comment by nilo19** [comment]
> Please do not update vendor directly. Follow guidelines in https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-update-ca-dependencies-particularly-k8siokubernetes

thanks for the information, but I do use `update-vendor.sh` to update the vendor directory in this pr.

**Comment by feiskyer** [comment]
@losipiuk  could you have a look again?

**Comment by losipiuk** [comment]
Thanks. I would prefer that commit that updates vendor and Azure changes were separate. But given fact we are on development branch (master) I believe we can live with those squashed.

/lgtm
/approve
